addon/TelemetryRappor.jsm

```diff
@@ -318,44 +414,25 @@ var TelemetryRappor = {
    * @param {float} params.f - Value for probability f.
    * @param {float} params.p - Value for probability p.
    * @param {float} params.q - Value for probability q.
+   * @param {boolean} isSimulation - Boolean that indicates if the addon is run in a simulation.
+   * @param {int} simCohort - Cohort for the simulation.
    *
    * @return An object containing the cohort and the encoded value in hex.
    */
-  createReport(name, value, params) {
-    // Generate the RAPPOR secret. This secret never leaves the client.
-    let secret = null;
-    log.debug("HEYYY");
-    try {
-      secret = Services.prefs.getCharPref(PREF_RAPPOR_SECRET);
-      if (secret.length != 64) {
-        secret = null;
-      }
-    } catch (e) {
-      log.error("Error getting secret from prefs", e);
-    }
-    if (!secret) {
-      let randomArray = new Uint8Array(32);
-      crypto.getRandomValues(randomArray);
-      secret = bytesToHex(randomArray);
-      Services.prefs.setCharPref(PREF_RAPPOR_SECRET, secret);
-    }
-
-    // If we haven't self-selected a cohort yet for this measurement, then do so now,
-    // otherwise retrieve the cohort.
+  createReport(name, value, params, isSimulation = false, simCohort = null) {
+    let secret = getSecret(name);
     let cohort = null;
-    try {
-      cohort = Services.prefs.getIntPref(PREF_RAPPOR_PATH + name + ".cohort");
-    } catch (e) {
-      log.error("Error getting the cohort", e);
+    if (isSimulation) {
+      cohort = simCohort;
+    } else {
+      cohort = getCohort(name, params.cohorts);
     }
-    if (!cohort) {
-      cohort = Math.floor(getRandomFloat() * params.cohorts);
-      Services.prefs.setIntPref(PREF_RAPPOR_PATH + name + ".cohort", cohort);
-    }
-
+    let report = createReport(value, params.filterSize, params.numHashFunctions, params.p, params.q, params.f, cohort, secret, name);
     return {
-      cohort: cohort,
-      report: bytesToHex(createReport(value, params.filterSize, params.numHashFunctions, params.p, params.q, params.f, cohort, secret, name)),
+      bloom: bytesToHex(report.bloom),
```

Comment:
Please don't change the output format of this too much. Let's retain `report`. Change it to something like:

```
{
  internalPrr: ...,
  internalIrr: ...,
  internalBool: ...,
  cohort: cohort,
  report: report: bytesToHex(createReport(value, params.filterSize, params.numHashFunctions, params.p, params.q, params.f, cohort, secret, name)),
```

Reply:
The value of `internalIrr` and `report` would be the same. Naming this field just `irr`

Reply:
Then just leave it as `report` please

Reply:
Done